### DISCUSSION TITLE
URI Template Matching Regex Min,Max

### DISCUSSION THREAD

**Discussion by RafaelCavazza** [micronaut-projects/micronaut-core#5252] [Q&A]
Hi,

Is there a way to define a {min,max} size when using a regex with a path parameter? 

Something like: 
```/bar/{foo:[a-zA-Z0-9]{1,35}}```

The '{' and '}' characters are breaking the annotation processor.

**Comment by valhallarepo** [comment]
Same problem here.

**Comment by joao96neto** [comment]
same here

**Comment by graemerocher** [comment]
Not currently, you can apply a `@Min` `@Max` constraint on the parameter and handle the error accordingly.

**Reply by RafaelCavazza** [reply]
Hi @graemerocher ,

We have a specific use case where there are two possible routes diverging only by the parameter regex.

Something like:
```/bar/{id1:[a-zA-Z0-9]{10,25}}```
```/bar/{id2:[a-zA-Z0]{1,10}}```

If I understand correctly: 
If the regex is **valid** micronaut can decide the correct route, right?

Is there any other way we can achieve this behavior?

**Comment by lancerbh** [comment]
+1

**Comment by dan1els** [comment]
+1 I want the micronaut be able to distinguish the correct root when long id or uuid is in path

**Comment by mdeterman** [comment]
+1

**Comment by mmwinther** [comment]
We have this same problem. At the very least the docs should specify that only a _subset_ of regex is supported.

**Comment by TatyOko28** [comment]
The issue: curly braces `{` and `}` are used **both** for:
- delimiting a path parameter in the URI template (`{foo}`)
- expressing a regex quantifier (`{1,35}`)

Most annotation processors (Spring, JAX-RS, etc.) cannot distinguish between them, causing a parsing error.

## Solutions

### 1. Simple and universal solution: validate inside the controller method

Instead of putting the length constraint in the mapping regex, declare the parameter without a regex and validate the length manually.

**Spring MVC:**

```java
@GetMapping("/bar/{foo}")
public String getBar(@PathVariable String foo) {
    // Explicit validation
    if (foo == null || !foo.matches("[a-zA-Z0-9]+") || foo.length() < 1 || foo.length() > 35) {
        throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Invalid foo");
    }
    // continue...
    return "ok";
}
